src/dune_lang/package_constraint.ml

```diff
@@ -9,12 +9,27 @@ module Op = struct
     | Lt
     | Neq
 
-  let equal a b =
+  let compare a b =
     match a, b with
-    | Eq, Eq | Gte, Gte | Lte, Lte | Gt, Gt | Lt, Lt | Neq, Neq -> true
-    | _ -> false
+    | Eq, Eq -> Ordering.Eq
+    | Eq, _ -> Lt
+    | _, Eq -> Gt
+    | Gte, Gte -> Eq
+    | Gte, _ -> Lt
+    | _, Gte -> Gt
+    | Lte, Lte -> Eq
+    | Lte, _ -> Lt
+    | _, Lte -> Gt
+    | Gt, Gt -> Eq
+    | Gt, _ -> Lt
+    | _, Gt -> Gt
+    | Lt, Lt -> Eq
+    | Lt, _ -> Lt
+    | _, Lt -> Gt
+    | Neq, Neq -> Eq
```

Comment:
What is the semantics of this pattern match? Maybe writing down a comment on the ordering would be helpful, since I don't think there's a natural ordering for comparison operators.